eigen_pod_proofs.go

```diff
@@ -14,35 +14,35 @@ import (
 )
 
 const (
-	BEACON_STATE_ROOT_PREFIX            = "BEACON_STATE_ROOT_"
-	BEACON_STATE_TOP_LEVEL_ROOTS_PREFIX = "BEACON_STATE_TOP_LEVEL_ROOTS_"
-	VALIDATOR_TREE_PREFIX               = "VALIDATOR_TREE_"
-	MAX_ORACLE_STATE_CACHE_SIZE         = 2000000
+	MAX_ORACLE_STATE_CACHE_SIZE = 2000000
 )
 
 type EigenPodProofs struct {
-	chainID                       uint64
-	oracleStateRootCache          *expirable.LRU[uint64, phase0.Root]
-	oracleStateTopLevelRootsCache *expirable.LRU[uint64, *beacon.BeaconStateTopLevelRoots]
-	oracleStateValidatorTreeCache *expirable.LRU[uint64, [][]phase0.Root]
-	oracleStateCacheExpirySeconds int
+	chainID                               uint64
+	oracleStateRootCache                  *expirable.LRU[uint64, phase0.Root]
+	oracleStateTopLevelRootsCache         *expirable.LRU[uint64, *beacon.BeaconStateTopLevelRoots]
+	oracleStateValidatorTreeCache         *expirable.LRU[uint64, [][]phase0.Root]
+	oracleStateValidatorBalancesTreeCache *expirable.LRU[uint64, [][]phase0.Root]
+	oracleStateCacheExpirySeconds         int
 }
 
 func NewEigenPodProofs(chainID uint64, oracleStateCacheExpirySeconds int) (*EigenPodProofs, error) {
```

Comment:
I think the cache is no longer needed since we are not sticking with a single beacon state anymore. Also, this could grow unbounded and lead to crashes.